### DISCUSSION TITLE
cache volume namespacing key

### DISCUSSION THREAD

**Discussion by rajatjindal** [dagger/dagger#12214] [maintainers]
as a follow up of https://github.com/dagger/dagger/pull/8724#discussion_r1848219633, we want to possibly revisit later the key used for namespacing the cache volume.

originally the idea was to use module digest, but that would mean if change the module source, the new cache volume will be created, so as part of https://github.com/dagger/dagger/pull/8724 we are creating cache volume based on `modName` + `modPath`

This *may* bring up its own set of challenges, so keeping this issue to track this discussion so that it does not get lost after the other PR is merged.

**Comment by mihaelmiklec** [comment]
Is it possible to allow users to customize the namespace key? I.e., if we want to also always use `git.repo` and `git.branch` for namespacing.
